src/Symfony/Component/Validator/Constraints/Choice.php

```diff
@@ -49,4 +50,38 @@ public function getDefaultOption()
     {
         return 'choices';
     }
+
+    public function __construct(
+        $choices = null,
+        $callback = null,
+        bool $multiple = null,
+        bool $strict = null,
+        int $min = null,
+        int $max = null,
+        string $message = null,
+        string $multipleMessage = null,
+        string $minMessage = null,
+        string $maxMessage = null,
+        $groups = null,
+        $payload = null,
+        array $options = []
+    ) {
+        if (\is_array($choices) && \is_string(key($choices))) {
+            $options = array_merge($choices, $options);
```

Comment:
@derrabus I often use associative arrays as choices - using the translation key as choice key ; one reason would be to share this constant array with the `ChoiceType` `choices` option. It was completely ok with Annotations. But this compatibility rule for the new Attribute usage is now concidering my choices keys as constraint options and breaks my code. I get that this is required to support both Annotation and Attribute but this changes the way this constraint is working.

```php
// stupid example
abstract class FormatEnum
{
    public const ALL = [
        'enum.format.example1' => 'f1',
        'enum.format.example2' => 'd6',
        'enum.format.example3' => 'u9',
    ];
}

// Annotation: 👍🏻 
// @Choice(choices=FormatEnum::ALL)

// Attribute: 👎🏻 
// #[Choice(choices: FormatEnum::ALL)]
```

Reply:
Can you please open a bug for your issue?